File: backend/employees-managment-system/src/main/java/com/horus/employeesmanagmentsystem/model/Employee.java
package com.horus.employeesmanagmentsystem.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Generated;

import javax.persistence.*;
import java.io.Serializable;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor

@Entity
public class Employee implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY )
    @Column(nullable = false,updatable = false)
    private Long id;
    @Column(nullable = false)
    private String name;
    @Column(nullable = false)
    private String email;
    @Column(nullable = false)
    private String jobTitle;
    @Column(nullable = false)
    private String imageUrl;
    @Column(nullable = false,updatable = false)
    private String employeeCode;
}

File: frontend/employees-managment-system/src/app/model/Employee.ts
export class Employee
{
  id:number;
  name:string;
  email:string;
  jobTitle:string;
  imageUrl:string;
  employeeCode:string;


  constructor(id: number, name: string, email: string, jobTitle: string, imageUrl: string, employeeCode: string)
  {
    this.id = id;
    this.name = name;
    this.email = email;
    this.jobTitle = jobTitle;
    this.imageUrl = imageUrl;
    this.employeeCode = employeeCode;
  }

}
